charts/graphprotocol-indexer-agent/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,14 @@
+{{- if .Values.monitoring.enabled -}}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
```

Comment:
Is it possible to configure autodiscovering through prometheus annotations on the deployment?
(values are random)
```
        prometheus.io/scrape: 'true'
        prometheus.io/port: '{{ .Values.metrics.port }}'
        prometheus.io/path: '/metrics'
```

Reply:
Based on https://github.com/prometheus-operator/kube-prometheus/pull/16#issuecomment-305933103 it seems that prom-operator doesn't support annotation discovery.

Also I see no benefits of using annotations instead of CRD, by default.

We can add support of passing custom annotations throw values, if it's important to support that in chainstack infra

Reply:
Yep, let's add support of custom annotations

Reply:
We already had support of custom podAnnotations. In last commit I have added support of custom service annotations